homeassistant/components/fan/__init__.py

```diff
@@ -246,7 +246,7 @@ async def async_set_speed(self, speed: str):
             await self.async_turn_off()
             return
 
-        if speed in self.preset_modes:
+        if self.preset_modes and speed in self.preset_modes:
```

Comment:
Instead of this, we could change the return value of `preset_modes` to not allow returning `None`

Reply:
I think I would agree on that, but assuming that is out of scope for this PR?